pixl_core/src/core/project_config.py

```diff
@@ -0,0 +1,103 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Project-specific configuration for Pixl."""
+from __future__ import annotations
+
+import logging
+from enum import Enum
+from pathlib import Path
+from typing import Any, Optional
+
+import yaml
+from decouple import Config, RepositoryEmpty, RepositoryEnv
+from pydantic import BaseModel, validator
+
+# Make sure local .env file is loaded if it exists
+env_file = Path.cwd() / ".env"
+config = Config(RepositoryEnv(env_file)) if env_file.exists() else Config(RepositoryEmpty())
+PROJECT_CONFIGS_DIR = Path(config("PROJECT_CONFIGS_DIR"))
+
+logger = logging.getLogger(__name__)
+
+
+def load_project_config(project_slug: str) -> PixlConfig | Any:
+    """
+    Load configuration for a project based on its slug.
+    Project needs to have a corresponding yaml file in the `$PROJECT_CONFIGS_DIR` directory.
+    """
+    configpath = PROJECT_CONFIGS_DIR / f"{project_slug}.yaml"
+    logger.warning(f"Loading config for {project_slug} from {configpath}")  # noqa: G004
+    if not configpath.exists():
+        raise FileNotFoundError(f"No config for {project_slug}. Please submit PR and redeploy.")  # noqa: EM102, TRY003
+    return _load_project_config(configpath)
+
+
+def _load_project_config(filename: Path) -> PixlConfig | Any:
+    """
+    Load configuration from a yaml file.
+    :param filename: Path to the yaml file
+    """
+    yaml_data = yaml.safe_load(filename.read_text())
+    return PixlConfig.model_validate(yaml_data)
+
+
+class _Project(BaseModel):
+    name: str
+    azure_kv_alias: Optional[str]
+    modalities: list[str]
+
+
+class _DestinationEnum(str, Enum):
+    """Defines the valid upload destinations."""
+
+    none = "none"
+    ftps = "ftps"
+    azure = "azure"
+    dicomweb = "dicomweb"
+
+
+class _Destination(BaseModel):
+    dicom: _DestinationEnum
+    parquet: _DestinationEnum
+
+    @validator("parquet")
```

Comment:
`validator` is deprecated as of pydantic v2 (supported on python >=3.8), so I would prefer to not add it and replace with `field_validator`. We had a separate discussion with @milanmlft as he gets a clash with some other library that I don't. I also couldn't find where we set the pydantic depdndency (and therefore version), perhaps we can do some pair programming to figure this one out.